### ISSUE TITLE
Add acyclic behavior

### ISSUE THREAD

**Issue by capsulecorplab** [capsulecorplab/mindgraph#1]
i.e., Tasks shouldn't contain closed self-referencing paths.

Not sure how to do this besides with a recursive function.

**Comment by rra94** [comment]
Is this an undirected graph? If so you can run a DFS and keep only the edges used by the DFS tree.

**Comment by capsulecorplab** [comment]
@rra94 yes, it is undirected. For context, I suppose I should also mention this issue mostly concerns the `append()` method on a `Node` object
